xwiki-platform-core/xwiki-platform-oldcore/src/main/java/org/xwiki/store/hibernate/AbstractHibernateAdapter.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.store.hibernate;
+
+import java.util.EnumSet;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.boot.Metadata;
+import org.hibernate.dialect.Dialect;
+import org.hibernate.mapping.PersistentClass;
+import org.hibernate.mapping.Table;
+import org.hibernate.tool.hbm2ddl.SchemaUpdate;
+import org.hibernate.tool.schema.TargetType;
+import org.slf4j.Logger;
+import org.xwiki.stability.Unstable;
+import org.xwiki.wiki.descriptor.WikiDescriptorManager;
+
+import com.xpn.xwiki.internal.store.hibernate.HibernateConfiguration;
+import com.xpn.xwiki.internal.store.hibernate.HibernateStore;
+
+/**
+ * A base implementation of {@link HibernateAdapter} with a default implementation for each API.
+ * 
+ * @version $Id$
+ * @since 17.1.0RC1
+ */
+@Unstable
+public abstract class AbstractHibernateAdapter implements HibernateAdapter
+{
+    private static final String VIRTUAL_MODE_SCHEMA = "schema";
+
+    @Inject
+    protected WikiDescriptorManager wikis;
+
+    @Inject
+    protected Logger logger;
+
+    @Inject
+    private HibernateStore hibernateStore;
+
+    @Inject
+    private HibernateConfiguration hibernateConfiguration;
+
+    // Configuration
+
+    @Override
+    public boolean isConfiguredInSchemaMode()
+    {
+        String virtualModePropertyValue = this.hibernateStore.getConfiguration().getProperty("xwiki.virtual_mode");
+        if (virtualModePropertyValue == null) {
+            virtualModePropertyValue = VIRTUAL_MODE_SCHEMA;
+        }
+        return StringUtils.equals(virtualModePropertyValue, VIRTUAL_MODE_SCHEMA);
+    }
+
+    @Override
+    public String getDatabaseFromWikiName()
+    {
+        return getDatabaseFromWikiName(this.wikis.getCurrentWikiId());
+    }
+
+    @Override
+    public String getDatabaseFromWikiName(String wikiId)
+    {
+        if (wikiId == null) {
```

Comment:
I know it's not really new code, but might worth having a check also for empty string no?

Reply:
From what I can see the main reason why this if is there is for the following code to not fail or return result that don't make sense (like a "null" string), but you should not have this kind of problem with en empty string.

I prefer to not take any chances for now.